File: Rod Cutting Problem/rodCuttingProblem.java
import java.io.*;
import java.lang.*;
import java.util.*;


class Main{
    //This function returns the maximum price that can be obtained on cutting the problems.
    public static int rodCuttingProblem(int price[],int n){
        int value[] = new int[n+1];
        value[0] = 0;
  
        // Build the table val[] in bottom up manner and return
        // the last entry from the table
        for (int i = 1; i<=n; i++){
            int max_val = Integer.MIN_VALUE;
            for (int j = 0; j < i; j++)max_val = Math.max(max_val, price[j] + value[i-j-1]);
            value[i] = max_val;
        }
  
        return value[n];
    }
    //Main Function
    public static void main(String args[]){
        int[] lengths = {1, 2, 3, 4, 6, 7, 8};
        int prices[] = {1, 5, 8, 9, 10, 17, 17, 20};
        int n = 4;
        System.out.println(rodCuttingProblem(prices, n));
    }
}

File: Rod Cutting Problem/rodCuttingProblem.cpp
#include<bits/stdc++.h>
using namespace std;

// Function that returns the best price for a rod of length n

int cutRod(int price[], int N){
    
   int dp[N+1];
   dp[0] = 0;
   int i, j;
  
   // Build the  dp table val[] in bottom up manner 
   
   for (i = 1; i<= N; i++){
       
       dp[i] = INT_MIN;
       
       for (j = 0; j < i; j++)
         dp[i] = max(dp[i], price[j] + dp[i-j-1]);
        
   }
  
   // Return the maximum price
   return dp[N];
}


int main(){
    
    int N = 8; // Size of Rod
    
	int price [] = {1, 5, 8, 9, 10, 17, 17, 20}; // Price for different pieces
	
	cout<<cutRod(price, N) <<"\n";
	
	return 0;
}
